### PR TITLE
feat(sdk-core): propagate reqId for prebuildTransaction downstream methods

### PR DESCRIPTION
**Pull request by bdesoky**
TICKET: WP-1862

## Description

Added `reqId` as a prameter to methods and fucntions that are called downstrea of `prebuildTransaction`. I've also updated `prebuildTransaction` to return the request tracer `reqId`.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I have added unit tests where possible to make sure that the api's `setRequestTracer` method is being called properly. I have also tested that `reqId` is returned from `prebuildTransaction`

### CODE REVIEW

File: modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts [hunk 1]
```diff
@@ -210,7 +212,8 @@ export async function offerUserToBitgoRShare(
   vssProof?: string,
   privateShareProof?: string,
   userPublicGpgKey?: string,
-  publicShare?: string
+  publicShare?: string,
+  reqId?: IRequestTracer
```
Comment (margueriteblair) [inline modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts:216]: Do we not have to add the `reqId` param to `ecdsa.ts` as well?

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts [hunk 2]
```diff
@@ -518,18 +521,23 @@ export class EddsaUtils extends baseTSSUtils<KeyShare> {
       txRequestId,
       rShare,
       encryptedSignerShare.share,
-      apiVersion
+      apiVersion,
+      undefined,
```
Comment (margueriteblair) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:525]: Maybe there's a way to clean these up so they're not all undefined? Maybe we can put `reqId` into a `params` object {} so we don't have to pass in 4 undefined's before?

File: modules/bitgo/test/v2/unit/tss/eddsa.ts [hunk 3]
```diff
@@ -536,6 +538,39 @@ describe('test tss helper functions', function () {
           'signerShare'
         ).should.be.rejectedWith('Invalid RShare, is not from User to Bitgo');
       });
+
+      it('should call setRequestTracer', async function () {
+        const signatureShare = {
```
Comment (margueriteblair) [inline modules/bitgo/test/v2/unit/tss/eddsa.ts:543]: tests look solid

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts (hunk 2 above)
Comment (bdesoky) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:525]: Yea I also don't like all those `undefined`, initially I thought about just having a `params` object with all of the parameters, but that would be breaking. If we put `reqId` into a `params` object, wouldn't we still need to use `undefined` for the previous parameters? Or do you mean putting all the optional parameters into `params`?

Comment (margueriteblair) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:525]: Yeah, you're right that would be breaking...hmmmmm. I was thinking passing in all those optionals as a param, but in that case for now this is fine

File: modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts (hunk 1 above)
Comment (bdesoky) [reply modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts:216]: Yes I think I missed those, will add them now

**Review by margueriteblair — APPROVED** [review]

**Pull request merged**